### ISSUE TITLE
Exclude code with regex

### ISSUE THREAD

**Issue by ZHeadlee** [stryker-mutator/stryker-js#2966]
**Is your feature request related to a problem? Please describe.**
My project uses material-ui's style engine, so the CSS is included in the JS files. Stryker mutates the CSS, but this doesn't affect the functionality of the unit tests at all, so it is not useful. I have a similar issue with the `prop-types` library which does runtime type checking. Mutating this section also doesn't do anything, so I would like to exclude it.

**Describe the solution you'd like**
I would like to be able to exclude some of the code in a file from mutation specified via RegEx, for example:

Exclude the "makeStyles" CSS engine section of my code which looks like this:

```
const useStyles = makeStyles((theme) => ({
  button: {
    margin: theme.spacing(1),
  },
  buttonContainer: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  caption: {
    textAlign: 'center',
    marginTop: '15%',
  },
  textField: { margin: theme.spacing(1) },
}));
```

with this expression: `/.*makeStyles\(\((.|\n)*\}\)\).*/`

**Describe alternatives you've considered**
I have considered using the *mutation range* config to specify what lines to mutate, but this would be tedious to do for my entire project, and my team would have to remember to update the lines any time the code is changed.

**Comment by nicojs** [comment]
Yeah, I've been thinking about this a lot. Would #1472 be something that could work for you? I think we would need a more robust solution in the future, but #1472 is being worked on a.t.m.

**Comment by nicojs** [comment]
Ignore-plugins are released in StrykerJS 7.3 🎉 . Ignoring something like `makeStyles` should be pretty easy, if you want help please don't hesitate to ask. 

https://twitter.com/stryker_mutator/status/1713557660832792870

https://stryker-mutator.io/docs/stryker-js/disable-mutants/#using-an-ignore-plugin
